src/lib/orionld/db/dbEntityTypesGet.cpp

```diff
@@ -176,5 +177,19 @@ KjNode* dbEntityTypesGet(OrionldProblemDetails* pdP)
     nodeP = next;
   }
 
-  return sortedArrayP;
+  char entityTypesId[64];
+  strncpy(entityTypesId, "urn:ngsi-ld:EntityTypeList:", sizeof(entityTypesId));
+  uuidGenerate(&entityTypesId[27]);
+
+  KjNode* typeNodeResponseP = kjObject(orionldState.kjsonP, NULL);
+  KjNode* idNodeP           = kjString(orionldState.kjsonP, "id", entityTypesId);
+  KjNode* typeNodeP         = kjString(orionldState.kjsonP, "type", "EntityTypeList");
+  KjNode* typeNodeListP     = kjArray(orionldState.kjsonP,  "typeList");
+
+  kjChildAdd(typeNodeListP, sortedArrayP);
+  kjChildAdd(typeNodeResponseP, idNodeP);
+  kjChildAdd(typeNodeResponseP, typeNodeP);
+  kjChildAdd(typeNodeResponseP, typeNodeListP);
+
```

Comment:
This is weird ... I don't see the usage of "sortedArrayP" ...
How is "typeNodeListP" filled in?

Perhaps you should remove the creation and addition of typeNodeListP to typeNodeResponseP and just add sortedArrayP to typeNodeResponseP?

Reply:
Fixed in 068159b18